### DISCUSSION TITLE
Workflow: Predictions/Forecasting during batch execution (unfinished batch)

### DISCUSSION THREAD

**Discussion by nielsvth** [salvadorgarciamunoz/pyphi#19] [Q&A] [answered]
Using the monitoring/predict functionalities of the phibatch module one can forecast the batch trajectory using the PCA/PLS model for 'unseen' test batches. But this only seems to work with completed batches.

The workflow works fine with test batches which have full data, but is there a way to perform this monitoring/prediction/forecasting also for a batch that is still running (so 'on-the-Go')? (let's say, only data for first 20% of execution is available, as the batch is still running).

Currently my approach is to modify the dataset of an unfinished batch to have the expected shape (basically adding nan data to match required, full batch sample shape).

**Comment by manutremo** [comment]
Hi, I apologize for not having a full answer for you, but I've been thinking the same for some time and have reached the same conclusion to try filling the rest of the dataset with nan - haven't tried it yet though.

**Comment by salvadorgarciamunoz** [comment, marked as answer]
Very sorry for not seeing this before. When you run the monitoring commands pyphibatch the routine actually mimics the process you are describing assuming first it only has the first sample, then the first two, etc, etc. you don't have to do it by hand.
